english/tests/tests.py

```diff
@@ -5,121 +5,292 @@
+            "end_time": "19:30:00"
+        }
+        url = reverse("teacher-classes-list-create")
         response = self.client.get(url)
         self.assertEqual(response.status_code, status.HTTP_200_OK)
         self.assertEqual(len(response.data), 1)
         self.assertDictEqual(dict(response.data[0]), expected_data)
 
-    def test_search_classes_with_filters(self):
+    def test_get_classes_failed(self):
+        self.client.force_login(user=self.student)
+        url = reverse("teacher-classes-list-create")
+        response = self.client.get(url)
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+    def test_search_classes_success(self):
+        self.client.force_login(user=self.teacher)
         expected_data = {
-            "pk": 3,
-            "name": "ThirdClass",
+            "pk": 1,
+            "name": "FirstClass",
             "students": [
-                {"pk": 2, "first_name": "Matthew", "last_name": "Becher"},
-                {"pk": 4, "first_name": "Brody", "last_name": "Nelson"},
-                {"pk": 5, "first_name": "Damian", "last_name": "Ward"},
                 {"pk": 1, "first_name": "Maksim", "last_name": "Lukash"},
+                {"pk": 3, "first_name": "Ayden", "last_name": "Henris"},
+                {"pk": 4, "first_name": "Damian", "last_name": "Ward"},
+                {"pk": 5, "first_name": "Jonah", "last_name": "Matthews"}
             ],
-            "teacher": {"pk": 3, "first_name": "Ayden", "last_name": "Henris"},
+            "teacher": {"pk": 2, "first_name": "Matthew", "last_name": "Becher"},
+            "days": ["Monday", "Wednesday", "Friday"],
+            "start_time": "18:00:00",
+            "end_time": "19:30:00"
         }
-        url = f'{reverse("classes-list")}?name=&teacher=3&students=5'
+        url = f'{reverse("classes-list")}?name=FirstClass&teacher=2&students=1'
         response = self.client.get(url)
         self.assertEqual(response.status_code, status.HTTP_200_OK)
         self.assertEqual(len(response.data), 1)
         self.assertDictEqual(dict(response.data[0]), expected_data)
 
+    def test_search_classes_failed(self):
+        self.client.force_login(user=self.teacher)
+        bad_url = f'{reverse("classes-list")}?name=LostClass'
+        empty_response = self.client.get(bad_url)
+        self.assertEqual(len(empty_response.data), 0)
 
-class LessonsListTest(APITestCase):
+    def test_get_class_success(self):
+        self.client.force_login(user=self.teacher)
+        url = "http://0.0.0.0:8000/english/teacher/classes/1/"
+        response = self.client.get(url)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+    def test_get_class_failed(self):
+        self.client.force_login(user=self.student)
+        url = "http://0.0.0.0:8000/english/teacher/classes/1/"
+        response = self.client.get(url)
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
```

Comment:
Need to add tests for updating classes: success and failed(403, incorrect start and end time)